bin/runYANGgenericstats.sh

```diff
@@ -129,212 +110,101 @@ if [ "$IS_PROD" = "True" ]; then
-      meta="IOS XE"
-      os="XE"
-      ((running = running + 1))
-      git=${path##*/cisco/xe/}
-      slash_removed=${git%/}
-      prefix=${slash_removed#*/}
-      prefix2=$(echo $prefix | tr -cd '[:alnum:]')
-      (python yangGeneric.py --allinclusive True --metadata "Cisco $meta $prefix from https://github.com/YangModels/yang/tree/main/vendor/cisco/xe/$git" --lint --prefix Cisco$os$prefix2 --rootdir "$path" >>$LOG 2>&1) &
-      PIDSXE+=("$!")
-      wait_for_processes "${PIDSXE[@]}"
-   done
-   # Wait for all child-processes until move to next OS
-   for PID in ${PIDSXE[@]}; do
-      wait $PID || exit 1
-   done
-
-   # Cisco XR
-   date +"%c: processing all Cisco XR modules " >>$LOG
-   declare -a PIDSXR
-   running=0
-   for path in $(ls -d $NONIETFDIR/yangmodels/yang/vendor/cisco/xr/*/); do
-      meta="IOS XR"
-      os="XR"
-      ((running = running + 1))
-      git=${path##*/cisco/xr/}
-      slash_removed=${git%/}
-      prefix=${slash_removed#*/}
-      prefix2=$(echo $prefix | tr -cd '[:alnum:]')
-      (python yangGeneric.py --allinclusive True --metadata "Cisco $meta $prefix from https://github.com/YangModels/yang/tree/main/vendor/cisco/xr/$git" --lint --prefix Cisco$os$prefix2 --rootdir "$path" >>$LOG 2>&1) &
-      PIDSXR+=("$!")
-      wait_for_processes "${PIDSXR[@]}"
-   done
-   # Wait for all child-processes until move to next OS
-   for PID in ${PIDSXR[@]}; do
-      wait $PID || exit 1
-   done
-
-   # Cisco SVO
-   date +"%c: processing all Cisco SVO modules " >>$LOG
-   declare -a PIDSSVO
-   running=0
-   for path in $(ls -d $NONIETFDIR/yangmodels/yang/vendor/cisco/svo/*/); do
-      meta="NCS"
-      os="SVO"
-      ((running = running + 1))
-      git=${path##*/cisco/svo/}
-      slash_removed=${git%/}
-      prefix=${slash_removed#*/}
-      prefix2=$(echo $prefix | tr -cd '[:alnum:]')
-      (python yangGeneric.py --allinclusive True --metadata "Cisco $meta $prefix from https://github.com/YangModels/yang/tree/main/vendor/cisco/svo/$git" --lint --prefix Cisco$os$prefix2 --rootdir "$path" >>$LOG 2>&1) &
-      PIDSSVO+=("$!")
-      wait_for_processes "${PIDSSVO[@]}"
-   done
-   # Wait for all child-processes until move to next vendor
-   for PID in ${PIDSSVO[@]}; do
-      wait $PID || exit 1
-   done
+   cisco () {
+      # syntax: cisco meta os
+      local meta=$1
+      local os_upper=$2
+      local os_lower=${os_lower,,}
+      date +"%c: processing all Cisco $os_upper modules " >>$LOG
+      for path in $(ls -d $NONIETFDIR/yangmodels/yang/vendor/cisco/$os_lower/*/); do
+         git=${path##*/cisco/$os_lower/}
+         slash_removed=${git%/}
+         prefix=${slash_removed#*/}
+         prefix2=$(echo $prefix | tr -cd '[:alnum:]')
+         yang_generic --allinclusive --metadata "Cisco $meta $prefix from https://github.com/YangModels/yang/tree/main/vendor/cisco/$os_lower/$git" --lint --prefix Cisco$os_upper$prefix2 --rootdir "$path"
```

Comment:
This doesn't seem to work properly:
rootdir: `/var/yang/nonietf/yangmodels/yang/vendor/cisco//nx/`
metadata: `Cisco NX OS nx from https://github.com/YangModels/yang/tree/main/vendor/cisco//nx/`
prefix: `CiscoNXnx`